apps/dashboard/src/components/side-navigation/side-navigation.tsx

```diff
@@ -28,14 +28,22 @@ const linkVariants = cva(
 
 type NavLinkProps = {
   to?: string;
+  modal?: (...args: any[]) => JSX.Element;
   isExternal?: boolean;
   className?: string;
   children: React.ReactNode;
 } & VariantProps<typeof linkVariants>;
 
-const NavLink = ({ to, isExternal, className, variant, children }: NavLinkProps) => {
+const NavLink = ({ to, isExternal, className, variant, modal: Modal, children }: NavLinkProps) => {
   const classNames = cn(linkVariants({ variant, className }));
   if (!to) {
+    if (Modal) {
+      return (
+        <Modal>
+          <span className={classNames}>{children}</span>
+        </Modal>
+      );
+    }
```

Comment:
added ability to render modal when clicking on the nav item

Reply:
I'm not sure I like this abstraction. Can't we just use `asChild` and render the button that opens the corresponding modal when necessary?

Reply:
We can, but we want to render fully custom modals for each link. They might differ in header, content, and footer docs link. If you know how to do this better, please suggest.